module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,17.67,True,2023-01-10T07:00:23,,
cowidev.vax.batch.australia,134.83,True,2023-01-10T07:02:21,,
cowidev.vax.batch.austria,16.04,True,2023-01-10T07:00:22,,
cowidev.vax.batch.belgium,9.2,True,2023-01-10T07:00:15,,
cowidev.vax.batch.bolivia,4.98,True,2023-01-10T07:00:11,,
cowidev.vax.batch.canada,33.73,True,2023-01-10T07:00:39,,
cowidev.vax.batch.chile,1.45,True,2023-01-10T07:00:07,,
cowidev.vax.batch.czechia,34.54,True,2023-01-10T07:00:40,,
cowidev.vax.batch.denmark,136.1,True,2023-01-10T07:02:22,,
cowidev.vax.batch.ecdc,139.78,True,2023-01-10T07:02:25,,
cowidev.vax.batch.ecuador,,,2023-01-10T07:00:06,,
cowidev.vax.batch.estonia,0.57,True,2023-01-10T07:00:06,,
cowidev.vax.batch.france,7.23,True,2023-01-10T07:00:13,,
cowidev.vax.batch.germany,3.54,True,2023-01-10T07:00:09,,
cowidev.vax.batch.greece,21.3,True,2023-01-10T07:00:27,,
cowidev.vax.batch.hong_kong,22.5,True,2023-01-10T07:00:28,,
cowidev.vax.batch.indonesia,,,2023-01-10T07:00:06,,
cowidev.vax.batch.ireland,0.65,True,2023-01-10T10:36:54,,
cowidev.vax.batch.israel,17.27,True,2023-01-10T07:00:23,,
cowidev.vax.batch.italy,8.66,True,2023-01-10T07:00:14,,
cowidev.vax.batch.japan,58.18,True,2023-01-10T07:01:04,,
cowidev.vax.batch.jersey,5.27,True,2023-01-10T07:00:11,,
cowidev.vax.batch.latvia,71.47,True,2023-01-10T07:01:17,,
cowidev.vax.batch.lithuania,5.26,True,2023-01-10T07:00:11,,
cowidev.vax.batch.luxembourg,3.08,True,2023-01-10T07:00:09,,
cowidev.vax.batch.malaysia,0.91,True,2023-01-10T07:00:07,,
cowidev.vax.batch.malta,1.46,True,2023-01-10T07:00:07,,
cowidev.vax.batch.netherlands,,,2023-01-10T07:00:06,,
cowidev.vax.batch.new_zealand,15.91,True,2023-01-10T07:00:22,,
cowidev.vax.batch.norway,0.6,True,2023-01-10T07:00:06,,
cowidev.vax.batch.peru,6.06,True,2023-01-10T07:00:12,,
cowidev.vax.batch.portugal,,,2023-01-10T07:00:06,,
cowidev.vax.batch.romania,65.17,True,2023-01-10T07:01:11,,
cowidev.vax.batch.saudi_arabia,16.43,True,2023-01-10T07:00:22,,
cowidev.vax.batch.singapore,6.12,True,2023-01-10T07:00:12,,
cowidev.vax.batch.slovakia,123.26,True,2023-01-10T07:02:10,,
cowidev.vax.batch.slovenia,3.42,True,2023-01-10T07:00:10,,
cowidev.vax.batch.south_korea,135.2,True,2023-01-10T07:02:22,,
cowidev.vax.batch.spc,26.37,True,2023-01-10T07:00:34,,
cowidev.vax.batch.sweden,21.04,True,2023-01-10T07:00:28,,
cowidev.vax.batch.switzerland,135.29,True,2023-01-10T07:02:24,,
cowidev.vax.batch.trinidad_and_tobago,3.05,True,2023-01-10T07:00:12,,
cowidev.vax.batch.ukraine,,,2023-01-10T07:00:10,,
cowidev.vax.batch.united_kingdom,17.69,True,2023-01-10T07:00:27,,
cowidev.vax.batch.united_states,135.61,True,2023-01-10T07:02:26,,
cowidev.vax.batch.uruguay,10.6,True,2023-01-10T07:00:22,,
cowidev.vax.batch.zimbabwe,3.44,True,2023-01-10T07:00:14,,
cowidev.vax.incremental.africacdc,,,2023-01-10T07:00:12,,
cowidev.vax.incremental.antigua_barbuda,,,2023-01-10T07:00:12,,
cowidev.vax.incremental.aruba,1.23,True,2023-01-10T07:00:13,,
cowidev.vax.incremental.azerbaijan,,,2023-01-10T07:00:12,,
cowidev.vax.incremental.bahrain,,,2023-01-10T07:00:12,,
cowidev.vax.incremental.bangladesh,41.57,True,2023-01-10T07:00:53,,
cowidev.vax.incremental.barbados,,,2023-01-10T07:00:12,,
cowidev.vax.incremental.brazil,0.77,True,2023-01-10T07:00:13,,
cowidev.vax.incremental.bulgaria,14.99,True,2023-01-10T07:00:27,,
cowidev.vax.incremental.china,34.49,True,2023-01-10T07:00:47,,
cowidev.vax.incremental.costa_rica,,,2023-01-10T07:00:13,,
cowidev.vax.incremental.croatia,1.3,True,2023-01-10T07:00:14,,
cowidev.vax.incremental.cuba,14.64,True,2023-01-10T07:00:28,,
cowidev.vax.incremental.curacao,1.15,True,2023-01-10T07:00:15,,
cowidev.vax.incremental.cyprus,4.78,True,2023-01-10T07:00:19,,
cowidev.vax.incremental.dominican_republic,12.19,True,2023-01-10T10:32:43,,
cowidev.vax.incremental.el_salvador,15.43,True,2023-01-10T07:00:30,,
cowidev.vax.incremental.faeroe_islands,2.22,True,2023-01-10T07:00:18,,
cowidev.vax.incremental.fiji,0.0,True,2023-01-10T07:00:18,,
cowidev.vax.incremental.finland,0.99,True,2023-01-10T07:00:19,,
cowidev.vax.incremental.gabon,,,2023-01-10T07:00:19,,
cowidev.vax.incremental.georgia,8.88,True,2023-01-10T07:00:28,,
cowidev.vax.incremental.greenland,0.2,False,2023-01-10T07:04:19,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 46, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/greenland.py"", line 76, in main
    Greenland().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/greenland.py"", line 63, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/greenland.py"", line 17, in read
    data = self._parse_data(soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/greenland.py"", line 21, in _parse_data
    return {**self._parse_data_metrics(soup), **self._parse_data_date(soup)}
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/greenland.py"", line 31, in _parse_data_metrics
    assert len(counter_1) == len(counter_2) == 1
AssertionError
",
cowidev.vax.incremental.guatemala,,,2023-01-10T07:00:22,,
cowidev.vax.incremental.guernsey,10.82,True,2023-01-10T07:00:32,,
cowidev.vax.incremental.hungary,124.1,True,2023-01-10T07:02:26,,
cowidev.vax.incremental.iceland,4.74,True,2023-01-10T07:00:26,,
cowidev.vax.incremental.india,3.66,True,2023-01-10T07:00:26,,
cowidev.vax.incremental.iran,,,2023-01-10T07:00:23,,
cowidev.vax.incremental.isle_of_man,2.38,True,2023-01-10T07:00:25,,
cowidev.vax.incremental.jamaica,9.91,True,2023-01-10T07:00:33,,
cowidev.vax.incremental.kazakhstan,7.49,True,2023-01-10T07:00:33,,
cowidev.vax.incremental.kosovo,6.12,False,2023-01-10T07:04:26,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 46, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/kosovo.py"", line 82, in main
    Kosovo().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/kosovo.py"", line 77, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/kosovo.py"", line 23, in read
    df = self._parse_data(soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/kosovo.py"", line 32, in _parse_data
    soup = get_soup(link, verify=False)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 83, in get_soup
    response = get_response(source, request_method, use_proxy, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 52, in get_response
    raise ValueError(
ValueError: Source https://msh.rks-gov.net/10.230.0.57+9000/coviddashboard.html not reached! Error code 404 Not Found
",Source https://msh.rks-gov.net/10.230.0.57+9000/coviddashboard.html not reached! Error code 404 Not Found
cowidev.vax.incremental.kyrgyzstan,5.25,True,2023-01-10T07:00:32,,
cowidev.vax.incremental.laos,8.48,True,2023-01-10T07:00:35,,
cowidev.vax.incremental.macao,11.82,True,2023-01-10T07:00:39,,
cowidev.vax.incremental.moldova,,,2023-01-10T07:00:27,,
cowidev.vax.incremental.monaco,129.46,True,2023-01-10T07:02:37,,
cowidev.vax.incremental.mongolia,1.49,True,2023-01-10T07:00:29,,
cowidev.vax.incremental.montenegro,0.76,True,2023-01-10T07:00:28,,
cowidev.vax.incremental.morocco,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.myanmar,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.nepal,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.north_macedonia,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.northern_cyprus,5.95,True,2023-01-10T07:00:34,,
cowidev.vax.incremental.paho,75.99,True,2023-01-10T10:33:47,,
cowidev.vax.incremental.pakistan,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.philippines,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.poland,2.61,True,2023-01-10T07:00:31,,
cowidev.vax.incremental.qatar,,,2023-01-10T07:00:28,,
cowidev.vax.incremental.russia,53.55,True,2023-01-10T07:01:22,,
cowidev.vax.incremental.saint_lucia,,,2023-01-10T07:00:29,,
cowidev.vax.incremental.serbia,0.39,False,2023-01-10T07:06:44,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 46, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/serbia.py"", line 88, in main
    Serbia().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/serbia.py"", line 74, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/serbia.py"", line 23, in read
    soup = get_soup(self.source_url)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 83, in get_soup
    response = get_response(source, request_method, use_proxy, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 52, in get_response
    raise ValueError(
ValueError: Source https://vakcinacija.gov.rs/ not reached! Error code 503 Service Unavailable
",Source https://vakcinacija.gov.rs/ not reached! Error code 503 Service Unavailable
cowidev.vax.incremental.south_africa,,,2023-01-10T07:00:30,,
cowidev.vax.incremental.spain,32.9,True,2023-01-10T07:01:03,,
cowidev.vax.incremental.sri_lanka,,,2023-01-10T07:00:30,,
cowidev.vax.incremental.suriname,15.37,True,2023-01-10T07:00:46,,
cowidev.vax.incremental.taiwan,11.06,False,2023-01-10T07:06:55,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 46, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 175, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 161, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 35, in read
    df = self._parse_table(url_pdf)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 61, in _parse_table
    raise ValueError(""Table 1: format has changed!"")
ValueError: Table 1: format has changed!
",Table 1: format has changed!
cowidev.vax.incremental.thailand,7.85,True,2023-01-10T07:00:40,,
cowidev.vax.incremental.turkey,40.05,False,2023-01-10T07:07:35,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connection.py"", line 174, in _new_conn
    conn = connection.create_connection(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/util/connection.py"", line 95, in create_connection
    raise err
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/util/connection.py"", line 85, in create_connection
    sock.connect(sa)
TimeoutError: timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connectionpool.py"", line 703, in urlopen
    httplib_response = self._make_request(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connectionpool.py"", line 386, in _make_request
    self._validate_conn(conn)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connectionpool.py"", line 1042, in _validate_conn
    conn.connect()
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connection.py"", line 358, in connect
    self.sock = conn = self._new_conn()
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connection.py"", line 179, in _new_conn
    raise ConnectTimeoutError(
urllib3.exceptions.ConnectTimeoutError: (<urllib3.connection.HTTPSConnection object at 0x7f28f898f190>, 'Connection to covid19asi.saglik.gov.tr timed out. (connect timeout=20)')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/adapters.py"", line 440, in send
    resp = conn.urlopen(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/connectionpool.py"", line 787, in urlopen
    retries = retries.increment(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/urllib3/util/retry.py"", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='covid19asi.saglik.gov.tr', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f28f898f190>, 'Connection to covid19asi.saglik.gov.tr timed out. (connect timeout=20)'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 46, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/turkey.py"", line 60, in main
    data = read(source).pipe(pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/turkey.py"", line 19, in read
    soup = get_soup(source)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 83, in get_soup
    response = get_response(source, request_method, use_proxy, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 49, in get_response
    raise err
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/web/scraping.py"", line 43, in get_response
    response = requests.get(source, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/api.py"", line 75, in get
    return request('get', url, params=params, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/api.py"", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/sessions.py"", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/sessions.py"", line 645, in send
    r = adapter.send(request, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/requests/adapters.py"", line 507, in send
    raise ConnectTimeout(e, request=request)
requests.exceptions.ConnectTimeout: HTTPSConnectionPool(host='covid19asi.saglik.gov.tr', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f28f898f190>, 'Connection to covid19asi.saglik.gov.tr timed out. (connect timeout=20)'))
","HTTPSConnectionPool(host='covid19asi.saglik.gov.tr', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7f28f898f190>, 'Connection to covid19asi.saglik.gov.tr timed out. (connect timeout=20)'))"
cowidev.vax.incremental.united_arab_emirates,,,2023-01-10T07:00:33,,
cowidev.vax.incremental.vietnam,,,2023-01-10T07:00:33,,
cowidev.vax.incremental.who,77.6,True,2023-01-10T07:01:50,,
cowidev.vax.incremental.zambia,1.91,True,2023-01-10T07:00:35,,
